tika-parsers/src/main/resources/org/apache/tika/parser/captioning/tf/model_wrapper.py

```diff
@@ -0,0 +1,347 @@
+#  software distributed under the License is distributed on an
+#  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+#  KIND, either express or implied.  See the License for the
+#  specific language governing permissions and limitations
+#  under the License.
+
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os.path
+
+import tensorflow as tf
+from tensorflow.contrib.slim.python.slim.nets.inception_v3 import inception_v3_base
+
+slim = tf.contrib.slim
+
+
+class ModelWrapper(object):
+    """
+        Model wrapper class to perform image captioning with a ShowAndTellModel
+    """
+
+    def __init__(self):
+        super(ModelWrapper, self).__init__()
+
+    def build_graph(self, checkpoint_path):
+        """Builds the inference graph"""
+
+        tf.logging.info("Building model.")
+        ShowAndTellModel().build()
+        saver = tf.train.Saver()
+
+        return self._create_restore_fn(checkpoint_path, saver)
+
+    def _create_restore_fn(self, checkpoint_path, saver):
+        """Creates a function that restores a model from checkpoint file"""
+
+        if tf.gfile.IsDirectory(checkpoint_path):
+            checkpoint_path = tf.train.latest_checkpoint(checkpoint_path)
+            if not checkpoint_path:
+                raise ValueError("No checkpoint file found in: %s" % checkpoint_path)
+
+        def _restore_fn(sess):
+            tf.logging.info("Loading model from checkpoint: %s", checkpoint_path)
+            saver.restore(sess, checkpoint_path)
+            tf.logging.info("Successfully loaded checkpoint: %s",
+                            os.path.basename(checkpoint_path))
+
+        return _restore_fn
+
+    def feed_image(self, sess, encoded_image):
+        initial_state = sess.run(fetches="lstm/initial_state:0",
+                                 feed_dict={"image_feed:0": encoded_image})
+        return initial_state
+
+    def inference_step(self, sess, input_feed, state_feed):
+        softmax_output, state_output = sess.run(
+            fetches=["softmax:0", "lstm/state:0"],
+            feed_dict={
+                "input_feed:0": input_feed,
+                "lstm/state_feed:0": state_feed,
+            })
+        return softmax_output, state_output
+
+
+class ShowAndTellModel(object):
+    """
+        Image captioning implementation based on the paper,
+
+        "Show and Tell: A Neural Image Caption Generator"
+        Oriol Vinyals, Alexander Toshev, Samy Bengio, Dumitru Erhan
+
+        For more details, please visit : http://arxiv.org/abs/1411.4555
+    """
+
+    def __init__(self):
+
+        # scale used to initialize model variables
+        self.initializer_scale = 0.08
+
+        # dimensions of Inception v3 input images
+        self.image_height = 299
+        self.image_width = 299
+
+        # image format ("jpeg" or "png")
+        self.image_format = "jpeg"
```

Comment:
see, it's hard coded for now, I'm thinking of a better way to pass the image format into the comp' graph from c type.